src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,396 @@
+        )
+
+    if current_fastcore_version < packaging.version.Version(fastcore_min_version):
+        raise ImportError(
+            f"`push_to_hub_fastai` and `from_pretrained_fastai` require a fastcore>={fastcore_min_version} version, but you are using fastcore version {get_fastcore_version()} which is incompatible. Upgrade with `pip install fastcore==1.3.27`."
+        )
+
+
+def check_fastai_fastcore_pyproject_versions(
+    storage_folder: str,
+    fastai_min_version: Optional[str] = "2.4",
+    fastcore_min_version: Optional[str] = "1.3.27",
+):
+    """
+    Checks that the `pyproject.toml` file in the directory `storage_folder` has fastai and fastcore versions
+    that are compatible with `save_fastai_learner`, `from_pretrained_fastai` and `push_to_hub_fastai`.
+
+    Parameters:
+        storage_folder (:obj:`str`):
+            Folder to look for the `pyproject.toml` file.
+        fastai_min_version (:obj:`str`, `optional`):
+            The minimum fastai version supported.
+        fastcore_min_version (:obj:`str`, `optional`):
+            The minimum fastcore version supported.
+
+    Raises:
+        FileNotFoundError, ImportError
+
+    """
+
+    try:
+        import toml
+    except IndexError:
+        raise ImportError(
+            "`push_to_hub_fastai` and `from_pretrained_fastai` require the toml module. Install with `pip install toml`."
+        )
+
+    # Check that a `pyproject.toml` exists in the repository, and if so, get a list of required packages
+    try:
+        package_versions = toml.load(f"{storage_folder}/pyproject.toml")[
+            "build-system"
+        ]["requires"]
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            "There is no `pyproject.toml` in the repository that contains the fastai Learner. This is necessary to check that the `from_pretrained_fatai` function is compatible with the fastai and fastcore versions used in the model you want to load."
+        )
+
+    # Check that `pyproject.toml` contains versions for fastai and fastcore. If there is no version available, it throws an error
+    try:
+        fastai_version = str(
+            [package for package in package_versions if package.startswith("fastai")][0]
+        )
+    except IndexError:
+        raise ImportError(
+            "The repository does not have a fastai version specified in `pyproject.toml`."
+        )
+    try:
+        fastcore_version = str(
+            [package for package in package_versions if package.startswith("fastcore")][
+                0
+            ]
+        )
+    except IndexError:
+        raise ImportError(
+            "The repository does not have a fastcore version specified in `pyproject.toml`."
+        )
+
+    # Get the fastai and fastcore versions in `pyproject.toml`. If there is none, it means that it defaults to the highest version.
+    fastai_version = fastai_version.partition("=")[2]
+    fastcore_version = fastcore_version.partition("=")[2]
```

Comment:
TL;DR: Partition is easier to use for this particular case than split.

I chose `partition` because of the particular case when the `pyproject.toml` only indicate `fastai` (without a version like `fastai=2.0.0`). In this case, `partition` would output an empty value for position [2] of the tuple (example: `('fastai', '', '')`). If we had something like `fastai=2.0.0` then the position [2] of the tuple would be the version: `('fastai', '=', '2.0.0')`. In other words, the tuple always has three values where the postion [2] indicate the version (empty or not).

This is not the case for `split` where the output can vary in length: `['fastai', '2.0.0']` when there is an explicit version and `['fastai']` when there is not. The case you propose (`fastai_version.split("=")[1]`) would output an error when no version is explicitly mentioned because there is no element [1].